File: objetos/ejemplos/Typescript/promedio/src/Alumno.ts
export class Alumno{
    private notas: number[] = [];

    constructor(private nombre: string, private dni: number){}

    public promedio() {
        var acu = 0;
        for (const nota of this.notas) {
            acu += nota;
        }
        return acu / this.notas.length;
    }

    public add(nota: number){
        this.notas.push(nota);
    }

}

File: objetos/ejemplos/scala/promedios/src/main/scala/Alumno.scala
import scala.collection.mutable.ListBuffer

class Alumno (private var dni : Long,
              private var nombre : String) {

  private var notas = new ListBuffer[Integer]()
  
  def promedio = {
    var acu = 0.0
    for(nota <- notas) {
      acu = acu + nota
    }
    acu / this.notas.size
  }
  
  def addNota(nota: Integer): Unit = {
    this.notas.addOne(nota)
  }

}
